module/platform/Webots/src/Webots.cpp

```diff
@@ -0,0 +1,257 @@
+    on<Trigger<ServoTargets>>().then([this](const ServoTargets& commands) {
+        // Maybe keep the `ServoTarget`s we have not sent yet, instead of just overriding them.
+        to_send.motor_positions.clear();
+        to_send.motor_velocities.clear();
+        to_send.motor_torques.clear();
+
+        // Store each ServoTarget to send in the next lot
+        for (auto& command : commands.targets) {
+            MotorPosition position_msg;
+            position_msg.name     = command.id;
+            position_msg.position = command.position;
+            to_send.motor_positions.push_back(position_msg);
+
+            // TODO(cmurtagh) work out if gain is velocity or force
+            MotorVelocity velocity_msg;
+            velocity_msg.name     = command.id;
+            velocity_msg.velocity = command.gain;
+            to_send.motor_velocities.push_back(velocity_msg);
+
+            MotorTorque torque_msg;
+            torque_msg.name   = command.id;
+            torque_msg.torque = command.torque;
+            to_send.motor_torques.push_back(torque_msg);
+
+            // MotorPID ? Do we need to send this?
+        }
+    });
+}
+
+void Webots::send_player_details(const int& fd, const GlobalConfig& player_details) {
+    // TODO(cameron) resend if failes
+    std::vector<char> data = NUClear::util::serialise::Serialise<GlobalConfig>::serialise(player_details);
+    uint32_t N             = htonl(data.size());
+    send(fd, &N, sizeof(N), 0);
+    send(fd, data.data(), N, 0);
+}
+
+void Webots::translate_and_emit_sensor(const SensorMeasurements& sensor_measurements) {
+            // Read each field of msg, translate it to our protobuf and emit the data
+            auto sensor_data = std::make_unique<DarwinSensors>();
+
+
+            sensor_data->timestamp = sensor_measurements.time;  // Not sure if we want this or the timestamp on the received message.
+
+            // Vecotor3 is a neutron of 3 doubles
+
+            for (auto& position : sensor_measurements.position_sensors) {
+                // string name
+                // double valvue
+            }
+
+            for (auto& accelerometer : sensor_measurements.accelerometers) {
+                // string name
+                // Vector3 value
+            }
+
+            for (auto& gyro : sensor_measurements.gyros) {
+                // string name
+                // Vector3 value
+            }
+
+            for (auto& bumper : sensor_measurements.bumpers) {
+                // string name
+                // bool value
+            }
+
+            for (auto& force_3d : sensor_measurements.force3ds) {
+                // string name
+                // Vector3 value
+            }
+
+            for (auto& force_6d : sensor_measurements.force6ds) {
+                // string name
+                // Vector3 value
+            }
+
+            emit(sensor_data);
+
+            for (auto& camera : sensor_measurements.cameras) {
```

Comment:
Can we make all of these loop variables like this one `const`?
```suggestion
            for (const auto& camera : sensor_measurements.cameras) {
```

Reply:
As long as we have a cont iterator